lib/src/badge/Badge.tsx

```diff
@@ -1,47 +1,156 @@
+    background: CoreTokens.color_yellow_200,
+    text: CoreTokens.color_yellow_900,
+  },
+  purple: {
+    background: CoreTokens.color_purple_200,
+    text: CoreTokens.color_purple_900,
+  },
+};
+
+const sizeMap = {
+  small: {
+    height: "20px",
+    width: "20px",
+    minWidth: "20px",
+    fontSize: CoreTokens.type_scale_01,
+    borderRadius: CoreTokens.spacing_12,
+    iconSize: "14px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_4}`,
+    },
+  },
+  medium: {
+    height: "24px",
+    width: "24px",
+    minWidth: "24px",
+    fontSize: CoreTokens.type_scale_02,
+    borderRadius: CoreTokens.spacing_12,
+    iconSize: "16px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4} ${CoreTokens.spacing_8}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_4}`,
+    },
+  },
+  large: {
+    height: "32px",
+    width: "32px",
+    minWidth: "32px",
+    fontSize: CoreTokens.type_scale_04,
+    borderRadius: CoreTokens.spacing_24,
+    iconSize: "24px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4} ${CoreTokens.spacing_8}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_8}`,
+    },
+  },
+};
+
+const Label = ({ label, notificationLimit, size }) => {
+  return (
+    <LabelContainer size={size}>
+      {typeof label === "number" ? (label > notificationLimit ? "+" + notificationLimit : label) : label}
+    </LabelContainer>
+  );
+};
+
+const DxcBadge = ({
+  label,
+  title,
+  mode = "contextual",
+  color = "grey",
+  icon,
+  notificationLimit = 99,
+  size = "medium",
+}: BadgePropsType): JSX.Element => {
   return (
-    <ThemeProvider theme={colorsTheme.tabs}>
-      <StyledDxcBadge notificationText={notificationText} disabled={disabled}>
-        <span>{notificationText}</span>
-      </StyledDxcBadge>
-    </ThemeProvider>
+    <BadgeContainer label={label} mode={mode} color={color} size={size} title={title} aria-label={title}>
+      {(mode === "contextual" && (
+        <DxcFlex gap="0.125rem" alignItems="center">
+          {icon && (
+            <IconContainer size={size}>
+              {typeof icon === "string" ? <img src={icon} alt={title} /> : icon}
+            </IconContainer>
+          )}
+          <Label label={label} notificationLimit={notificationLimit} size={size} />
+        </DxcFlex>
+      )) || <Label label={label} notificationLimit={notificationLimit} size={size} />}
+    </BadgeContainer>
   );
 };
 
-const StyledDxcBadge = styled.div<{
-  notificationText: BadgePropsType["notificationText"];
-  disabled: BadgePropsType["disabled"];
+const getColor = (mode, color) => {
+  return mode === "contextual" ? contextualColorMap[color].text : CoreTokens.color_white;
+};
+
+const getBackgroundColor = (mode, color) => {
+  return mode === "contextual" ? contextualColorMap[color].background : CoreTokens.color_red_700;
+};
+
+const getPadding = (mode, size) => {
```

Comment:
In these three functions(getColor, getBackgroundColor, and getPadding) we can remove the braces and 'return' and leave them in a single line.

Reply:
Agree!